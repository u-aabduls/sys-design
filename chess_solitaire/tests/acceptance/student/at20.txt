-- ACCEPTANCE TEST #20: VIOLATION CASE: moves - slot_valid

-- Setup some chess pieces for the game
setup_chess (K,1,3)
setup_chess (K,2,4)

-- start game
start_game

-- check moves for (-1,3), an out of bounds
-- position on the game board
moves(-1,3)

-- check moves for (-9,-3), an out of bounds
-- position on the game board
moves(-9,-3)

-- check moves for (2,5), an out of bounds
-- position on the game board
moves(2,5)

-- check moves for (7,2), an out of bounds
-- position on the game board
moves(7,2)

-- check moves for (7,10), an out of bounds
-- position on the game board
moves(7,10)

