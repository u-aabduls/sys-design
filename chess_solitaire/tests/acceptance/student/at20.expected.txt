  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(K,1,3)
  # of chess pieces on board: 1
  Game being Setup...
  ..K.
  ....
  ....
  ....
->setup_chess(K,2,4)
  # of chess pieces on board: 2
  Game being Setup...
  ..K.
  ...K
  ....
  ....
->start_game
  # of chess pieces on board: 2
  Game In Progress...
  ..K.
  ...K
  ....
  ....
->moves(-1,3)
  # of chess pieces on board: 2
  Error: (-1, 3) not a valid slot
  ..K.
  ...K
  ....
  ....
->moves(-9,-3)
  # of chess pieces on board: 2
  Error: (-9, -3) not a valid slot
  ..K.
  ...K
  ....
  ....
->moves(2,5)
  # of chess pieces on board: 2
  Error: (2, 5) not a valid slot
  ..K.
  ...K
  ....
  ....
->moves(7,2)
  # of chess pieces on board: 2
  Error: (7, 2) not a valid slot
  ..K.
  ...K
  ....
  ....
->moves(7,10)
  # of chess pieces on board: 2
  Error: (7, 10) not a valid slot
  ..K.
  ...K
  ....
  ....
